Upgrade to Python 3.10

cfdm won't import under Python 3.10 (see #186). This issue needs fixing, as well as any other items that arise from testing:

* [x] Account for `__wrapper__` being a read-only attribute at Python 3.10 (fixed by #189)
* [x] Address following decorator-related(?) issue or test manifestation, resulting in this test error:

    ```python
    ======================================================================
    FAIL: test_manage_log_level_via_verbosity (test_decorators.decoratorsTest)
    Test the `_manage_log_level_via_verbosity` decorator.
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/sadie/cfdm/cfdm/test/test_decorators.py", line 204, in test_manage_log_level_via_verbosity
        self.assertIn(msg, catch.output)
    AssertionError: 'INFO:test_decorators:This should be short and sweet' not found in ['WARNING:test_decorators:Best pay attention to this!']
    ```
     (fixed by #197)

--- a/Changelog.rst
+++ b/Changelog.rst
@@ -19,6 +19,14 @@ Version 1.9.1.0
 
 ----
 
+Version 1.9.0.4
+---------------
+
+**2022-07-??**
+
+* Upgrade to allow cfdm to work with Python 3.10
+  (https://github.com/NCAS-CMS/cfdm/issues/187)
+
 Version 1.9.0.3
 ---------------
 

--- a/cfdm/core/meta/docstringrewrite.py
+++ b/cfdm/core/meta/docstringrewrite.py
@@ -197,10 +197,10 @@ class DocstringRewriteMeta(type):
             if is_classmethod:
                 attrs[attr_name] = classmethod(attr)
 
-            if is_staticmethod:
+            elif is_staticmethod:
                 attrs[attr_name] = staticmethod(attr)
 
-            if is_wrapped:
+            elif is_wrapped:
                 wrapper.__doc__ = attr.__doc__
                 wrapper.__wrapped__ = attr
                 attrs[attr_name] = wrapper

